File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String, Integer> map = new HashMap<String, Integer>();
        for(int i = 0; i<participant.length; i++){
            if(map.containsKey(participant[i])){
                int count = map.get(participant[i]);
                map.put(participant[i], ++count);
                
            }
            else{
                map.put(participant[i],1);
            }
        }
        for(String i:completion){
            int count = map.get(i);
            map.put(i,--count);
            if(map.get(i) == 0){
                map.remove(i);
            }
        }
        for(String j:participant){
            if(map.containsKey(j)){
                answer += (j);
                map.remove(j);
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dictionary = dict()
    for p in participant:
        if p in dictionary:
            dictionary[p] += 1
        else:
            dictionary[p] = 1

    for c in completion:
        dictionary[c] -= 1

    for k, v in dictionary.items():
        if v > 0:
            answer = k
    return answer